File: pset6/readability.py
from cs50 import get_string

# Get input text
text = get_string("Text: ")

# Count letters (A-Z and a-z)
letters = sum(1 for c in text if c.isalpha())

# Count words (assumes words are separated by spaces)
words = len(text.split())

# Count sentences (ends with ., ! or ?)
sentences = sum(1 for c in text if c in ['.', '!', '?'])

# Coleman-Liau index formula
L = (letters / words) * 100
S = (sentences / words) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

# Output result
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Get input from user
    string text = get_string("Text: ");

    // Count letters, words, sentences
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Calculate L and S (per 100 words)
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;

    // Coleman-Liau formula
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    // Output based on grade level
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

// Function to count alphabetic letters
int count_letters(string text)
{
    int count = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

// Function to count words (based on spaces)
int count_words(string text)
{
    int count = 1; // Start at 1 because first word has no space before it
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (isspace(text[i]))
        {
            count++;
        }
    }
    return count;
}

// Function to count sentences (ends with . or ! or ?)
int count_sentences(string text)
{
    int count = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}
